test/hotspot/jtreg/compiler/print/CompileCommandMemLimit.java

```diff
@@ -92,19 +88,38 @@ public static void main(String[] args) throws Exception {
         options.add("-XX:CompileCommand=compileonly," + getTestClass() + "::*");
 
         // We want a final report
-        options.add("-XX:CompileCommand=MemStat,*.*,print");
-
-        // We limit method 2 to a very small limit that is guaranteed to trigger
-        options.add("-XX:CompileCommand=MemLimit," + getTestMethod(METHOD2) + ",4k" + (test_crash ? "~crash" : ""));
-
-        // We disable any limit set on method 3
-        options.add("-XX:CompileCommand=MemLimit," + getTestMethod(METHOD3) + ",0");
-
-        if (c2) {
+        options.add("-XX:CompileCommand=MemStat," + getTestMethod(METHOD2) + ",print");
+
+        String suffix = mode == TestMode.crash ? "~crash" : "";
+
+        // About the limit:
+        //
+        // In the debug JVM, for this test class, compilers will allocate (near the very end of the compilation)
+        // 32MB of arena memory.
+        //
+        // C1 will allocate them in a single step from RA, leaked until end of compilation.
+        //
+        // C2 will allocate them in two steps: first 2MB inside testPhase 1 in a temporary arena that will be gone
+        // by phase end. So, in the phase timeline these 2MB must show up as "significant temporary peak".
+        // In testPhase2, we allocate 32MB from resource area, which is leaked until the end of the compilation. This
+        // means that these 32MB will show up as permanent memory increase in the phasetimeline.
```

Comment:
The references to `testPhase` seem obsolete, do you mean `Phase::_t_testTimer1` and `Phase::_t_testTimer2`?

Reply:
Right you are